projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java

```diff
@@ -43,15 +45,24 @@ public class DataJobsSynchronizer {
    * <p>This method retrieves job information from the database, compares it with the current state
    * of Kubernetes CronJobs in the cluster, and takes the necessary actions to synchronize them.
    * This can include creating new CronJobs, updating existing CronJobs, etc.
-   *
-   * @throws ApiException
    */
-  public void synchronizeDataJobs() throws ApiException {
+  public void synchronizeDataJobs() {
     ThreadPoolTaskExecutor taskExecutor = initializeTaskExecutor();
     Iterable<DataJob> dataJobsFromDB = jobsService.findAllDataJobs();
 
-    Set<String> dataJobDeploymentNamesFromKubernetes =
-        deploymentService.findAllActualDeploymentNamesFromKubernetes();
+    Set<String> dataJobDeploymentNamesFromKubernetes;
+    try {
+      dataJobDeploymentNamesFromKubernetes =
+          deploymentService.findAllActualDeploymentNamesFromKubernetes();
+    } catch (ApiException e) {
+      log.error(
+          "Skipping data job deployment synchronization because deployment names cannot be loaded"
+              + " from Kubernetes.",
+          e);
```

Comment:
Wrap this in KubernetesException. 

The ApiException message is really unhelpful - it doesn't print at all the content of the body or headers and it's very hard to understand what is the problem without those

Reply:
Done